martin-mbtiles/src/tile_copier.rs

```diff
@@ -6,37 +6,87 @@ use crate::{MbtError, Mbtiles};
 use sqlx::sqlite::{SqliteArguments, SqliteConnectOptions};
 use sqlx::{query, query_with, Arguments, Connection, Row, SqliteConnection};
 use std::collections::HashSet;
+use std::ffi::OsStr;
 use std::path::PathBuf;
+#[cfg(feature = "cli")]
+use {clap::error::ErrorKind, clap::Args};
 
 #[derive(Clone, Default, Debug)]
+#[cfg_attr(feature = "cli", derive(Args))]
 pub struct TileCopierOptions {
-    zooms: HashSet<u8>,
+    /// MBTiles file to read from
+    src_file: PathBuf,
+    /// MBTiles file to write to
+    dst_file: PathBuf,
+    /// Force the output file to be in a simple MBTiles format with a `tiles` table
+    #[cfg_attr(feature = "cli", arg(long))]
+    force_simple: bool,
+    /// Minimum zoom level to copy
+    #[cfg_attr(feature = "cli", arg(long))]
     min_zoom: Option<u8>,
+    /// Maximum zoom level to copy
+    #[cfg_attr(feature = "cli", arg(long))]
     max_zoom: Option<u8>,
-    //self.bbox = bbox
-    verbose: bool,
+    /// List of zoom levels to copy; if provided, min-zoom and max-zoom will be ignored
+    #[cfg_attr(feature = "cli", arg(long, value_parser(clap::builder::ValueParser::new(HashSetValueParser{}))))]
+    zoom_levels: HashSet<u8>,
+}
+
+#[cfg(feature = "cli")]
+#[derive(Clone)]
+struct HashSetValueParser;
+
+#[cfg(feature = "cli")]
+impl clap::builder::TypedValueParser for HashSetValueParser {
+    type Value = HashSet<u8>;
+
+    fn parse_ref(
+        &self,
+        _cmd: &clap::Command,
+        _arg: Option<&clap::Arg>,
+        value: &OsStr,
+    ) -> Result<Self::Value, clap::Error> {
+        let mut result = HashSet::<u8>::new();
```

Comment:
not certain here - do you actually have to specify the hashset type?  I would think Rust should figure it out based on the `.insert()` below